src/charm.py

```diff
@@ -95,19 +122,39 @@ def _on_install(self, event):
         else:
             self.unit.status = BlockedStatus("Missing token or org/repo path config")
 
-    def _on_upgrade_charm(self, event):
+    def _on_upgrade_charm(self, event: UpgradeCharmEvent) -> None:
+        """Handle the update of charm.
+
+        Args:
+            event (UpgradeCharmEvent): Event of charm upgrade.
+        """
         RunnerManager.install_deps()
 
-    def _on_config_changed(self, event):
-        self._ensure_event_timer("update-runner-bin", self.config["update-interval"])
-        self._ensure_event_timer("reconcile-runners", self.config["reconcile-interval"])
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+        """Handle the configuration change.
+
+        Args:
+            event (ConfigChangedEvent): Event of configuration change.
+        """
+        try:
+            self._event_timer.ensure_event_timer(
+                "update-runner-bin", self.config["update-interval"]
+            )
+            self._event_timer.ensure_event_timer(
+                "reconcile-runners", self.config["reconcile-interval"]
+            )
+        except TimerEnableError as ex:
+            logger.exception("Failed to start the event timer")
+            self.unit.status = BlockedStatus(
+                f"Failed to start timer for regular reconciliation and binary update checks: {ex}"
+            )
 
         if self.config["path"] != self._stored.path:
-            prev_runner_manager = self._get_runner_manager(path=self._stored.path)
+            prev_runner_manager = self._get_runner_manager(
+                path=str(self._stored.path)
+            )  # Casting for mypy checks.
```

Comment:
Out of curiosity, what is the type before the cast?

Reply:
`self._stored` => `BoundStoredState`
`self._stored.path` => `ObjectEvents`